src/ride/controllers/search_ride_controller.py

```diff
@@ -0,0 +1,29 @@
+from api.utils import failure_response
+from api.utils import success_response
+from api.utils import update
+from django.core.exceptions import ObjectDoesNotExist
+from person.models import Person
+
+from ..models import Ride
+from ..models import Path
+
+class SearchRideController:
+    def __init__(self, data, request, serializer): 
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        departure_datetime = self._data.get("departure_datetime")
+        start_location_name = self._data.get("start_location_name")
+        end_location_name = self._data.get("end_location_name")
+        if not (Path.objects.filter(start_location_name = start_location_name, end_location_name = end_location_name)).exists():
+            return failure_response("Path does not exist.")
```

Comment:
don't think we need a failure response for this, we can just return an empty list if we dont have that path yet

Reply:
+1 it will simplify frontend's job of having to handle a special case where nothing is found and they have to show nothing - instead give them an empty list so they just show nothing